host/xtest/regression_4000.c

```diff
@@ -980,6 +980,13 @@ static void xtest_tee_test_4001(ADBG_Case_t *c)
 		uint8_t out[64] = { };
 		size_t out_size = 0;
 
+		if (hash_cases[n].algo == TEE_ALG_SM3 &&
```

Comment:
Why are we only checking this for `TEE_ALG_SM3`? Because it's safer this way in case you don't read the log carefully?

Reply:
Yes. Since other algorithms are mandatory (as per GP), we probably want to make sure they have not been disabled by mistake.
On the other hand, if anyone configures a special version of OP-TEE without support for some algorithms, he will likely want to test it with `xtest` too -- that would require changing the code a bit.

Reply:
Agree.